ocs_ci/deployment/metallb.py

```diff
@@ -698,3 +706,110 @@ def apply_icsp(self):
+            resource_name=constants.METALLB,
+        )
+        metallb_catalog_source = CatalogSource(
+            resource_name=constants.QE_APP_REGISTRY_CATALOG_SOURCE_NAME,
+            namespace=MARKETPLACE_NAMESPACE,
+        )
+
+        logger.info("Check if metallb is installed")
+        if not self.metallb_instance_created():
+            logger.error("metalLb operator unavailable")
+            return False
+
+        logger.info(
+            f"Currently installed metallb version: {parse_version(self.get_running_metallb_version())}"
+        )
+        self.upgrade_version = config.UPGRADE.get("upgrade_metallb_version")
+        if not self.upgrade_version:
+            self.upgrade_version = get_running_ocp_version()
+        logger.info(
+            f"Upgarde metallb version to: {parse_version(self.upgrade_version)}"
+        )
+        if self.upgrade_version == parse_version(self.get_running_metallb_version()):
+            logger.info("Metallb operator is not upgradeable")
+            return True
+
+        self.catalog_source_name = constants.QE_APP_REGISTRY_CATALOG_SOURCE_NAME
+        catalog_source_data = templating.load_yaml(QE_APP_REGISTRY_SOURCE)
+        # create catsrc
+        if self.catalog_source_created():
+            logger.info(
+                f"Catalog Source {constants.QE_APP_REGISTRY_CATALOG_SOURCE_NAME} already exists"
+            )
+            # Upgrade image details in the catalogsource
+            image_placeholder = catalog_source_data.get("spec").get("image")
+            catalog_source_data.get("spec").update(
+                {"image": image_placeholder.format(self.upgrade_version)}
+            )
+            # install catalog source
+            metallb_catalog_file = tempfile.NamedTemporaryFile(
+                mode="w+", prefix="metallb_catalogsource", delete=False
+            )
+            templating.dump_data_to_temp_yaml(
+                catalog_source_data, metallb_catalog_file.name
+            )
+            disable_specific_source(constants.QE_APP_REGISTRY_CATALOG_SOURCE_NAME)
+            run_cmd(f"oc apply -f {metallb_catalog_file.name}", timeout=2400)
+
+            # wait for catalog source is ready
+            metallb_catalog_source.wait_for_state("READY")
+
+        else:
+            self.create_catalog_source(metallb_version=self.upgrade_version)
+
+        # wait for sometime before checking the latest metallb version
+        time.sleep(60)
+        if metallb_subs_obj.get()["spec"]["installPlanApproval"] != "Automatic":
+            patch = '{"spec": {"installPlanApproval": "Automatic"}}'
+            metallb_subs_obj.patch(params=patch, format_type="merge")
+            wait_for_install_plan_and_approve(self.namespace)
+
+        metallb_version_post_upgrade = self.get_running_metallb_version()
+        logger.info(f"metallb version post upgrade: {metallb_version_post_upgrade}")
+        return self.upgrade_version in metallb_version_post_upgrade
```

Comment:
Similarly as above, here it is not necessary to change the `installPlanApproval` back to `Manual` similarly to the cnv upgrade?